src/utlist.h

```diff
@@ -600,6 +632,41 @@ do {
   }                                                                                            \
 } while (0)
 
+#define DL_INSERT_INORDER(head,add,cmp)                                                        \
+  DL_INSERT_INORDER2(head,add,cmp,prev,next)
+
+#define DL_INSERT_INORDER2(head,add,cmp,prev,next)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
+    _CASTASGN(_pos, head);                                                                     \
+    if (head) {                                                                                \
+      DL_LOWER_BOUND(_pos,add,cmp);                                                            \
+      DL_APPEND_ELEM(head,_pos,add);                                                           \
+    }                                                                                          \
+    else {                                                                                     \
+      (head) = add;                                                                            \
+      (head)->next = NULL;                                                                     \
+      (head)->prev = NULL;                                                                     \
+    }                                                                                          \
+  } while (0)
+
+
+#define DL_LOWER_BOUND(head,node,cmp)                                                          \
+  DL_LOWER_BOUND2(head,node,cmp,prev,next)                                                     \
+
+#define DL_LOWER_BOUND2(head,node,cmp,prev,next)                                               \
+  do {                                                                                         \
+    if (cmp(head, node) > 0) {                                                                 \
+      (head) = NULL;                                                                           \
+    } else {                                                                                   \
+      for (; (head)->next != NULL; (head) = (head)->next ) {                                   \
+        if ((head)->next == (node)) break;                                                     \
```

Comment:
Why is line 663 needed/wanted? The `LL_` and `CDL_` versions of this code don't have this line.